File: app/src/main/java/com/commonpepper/photosen/ui/fragments/UserPhotosListFragment.java
package com.commonpepper.photosen.ui.fragments;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;

import com.commonpepper.photosen.ui.viewmodels.UserPhotosFragmentViewModelFactory;

public class UserPhotosListFragment extends AbstractListFragment {
    private static final String TAG_USER_ID = "user_id";

    public static UserPhotosListFragment newInstance(String user_id) {
        UserPhotosListFragment newFragment = new UserPhotosListFragment();
        Bundle args = new Bundle();
        args.putString(TAG_USER_ID, user_id);
        newFragment.setArguments(args);
        return newFragment;
    }

    public UserPhotosListFragment() {
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle args = getArguments();
        String user_id = args.getString(TAG_USER_ID);
        UserPhotosFragmentViewModelFactory factory = new UserPhotosFragmentViewModelFactory(user_id);

        mViewModel = ViewModelProviders.of(this, factory).get(UserPhotosFragmentViewModelFactory.UserPhotosListFragmentViewModel.class);
    }
}

File: app/src/main/java/com/commonpepper/photosen/ui/fragments/UserPhotosListFragment.kt
package com.commonpepper.photosen.ui.fragments

import android.os.Bundle
import androidx.lifecycle.ViewModelProviders
import com.commonpepper.photosen.ui.viewmodels.UserPhotosFragmentViewModelFactory
import com.commonpepper.photosen.ui.viewmodels.UserPhotosFragmentViewModelFactory.UserPhotosListFragmentViewModel

class UserPhotosListFragment : AbstractListFragment() {
    companion object {
        private const val TAG_USER_ID = "user_id"
        fun newInstance(user_id: String?): UserPhotosListFragment {
            val newFragment = UserPhotosListFragment()
            val args = Bundle()
            args.putString(TAG_USER_ID, user_id)
            newFragment.arguments = args
            return newFragment
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val args = arguments
        val userId = args?.getString(TAG_USER_ID) ?: ""
        val factory = UserPhotosFragmentViewModelFactory(userId)
        mViewModel = ViewModelProviders.of(this, factory).get(UserPhotosListFragmentViewModel::class.java)
    }
}